src/natstime.c

```diff
@@ -58,6 +58,33 @@ nats_NowInNanoSeconds(void)
 #endif
 }
 
+int64_t
+nats_NowMonotonicInNanoSeconds(void)
+{
+#ifdef _WIN32
+    LARGE_INTEGER frequency;
+    LARGE_INTEGER counter;
+    if (!QueryPerformanceFrequency(&frequency) || !QueryPerformanceCounter(&counter))
+        abort();
```

Comment:
This does not seem - somehow - to match the description you have in nats.h. If not available, the time is still expressed in nanoseconds, but seems to indicate that it would still return the non monotonic "now".

So I would flip that and say that if both ok, then use next line, but if not, then return `nats_NowInNanoSeconds()`.